### PR TITLE
Double quote --storage argument before passing it to raylet

### LINKED ISSUE #41568: default_worker.py  crashes when initializing --storage in ray head node
### What happened + What you expected to happen

Hello, 
I am trying to start ray head with following parameters:
`ray start --head  --dashboard-host=0.0.0.0 --block --storage="s3://explore/ray?&endpoint_override=http://192.168.0.4:9000"`

When I am trying to start serve with following code:

**main.py**
```
import ray
import time
from ray import serve
import logging

runtime_env = {
    "env_vars": {"TF_WARNINGS": "none"}
}

rayContext = ray.init(address='ray://ray_head_address_goes_here:10001', 
                      runtime_env=runtime_env,
                      logging_level=logging.DEBUG, log_to_driver=True
                    )

print('Starting Serve...')
serve.start(detached=True, http_options={"host": "0.0.0.0"})

while True:
    time.sleep(1)
```

**Translator.py**
```
from starlette.requests import Request
import ray
from ray import serve

app = FastAPI()

@serve.deployment(name="main", route_prefix="/")
@serve.ingress(app)
class Translator:
    def __init__(self):
        pass
      
    async def __call__(self, http_request: Request) -> str:
        return "hello"

```

default_worker.py crashes with output:
```
2023-11-30 00:01:26,715 DEBUG worker.py:646 -- Retaining 16310a0f0a45af5cb10accb0881646893aef9f9f0100000001000000
(raylet) bash: --startup-token=16: command not found
(raylet) usage: default_worker.py [-h] --cluster-id CLUSTER_ID --node-ip-address
(raylet)                          NODE_IP_ADDRESS --node-manager-port NODE_MANAGER_PORT
(raylet)                          [--raylet-ip-address RAYLET_IP_ADDRESS]
(raylet)                          --redis-address REDIS_ADDRESS --gcs-address
(raylet)                          GCS_ADDRESS [--redis-password REDIS_PASSWORD]
(raylet)                          --object-store-name OBJECT_STORE_NAME
(raylet)                          [--raylet-name RAYLET_NAME]
(raylet)                          [--logging-level {debug,info,warning,error,critical}]
(raylet)                          [--logging-format LOGGING_FORMAT]
(raylet)                          [--temp-dir TEMP_DIR] [--storage STORAGE]
(raylet)                          [--load-code-from-local] [--worker-type WORKER_TYPE]
(raylet)                          --metrics-agent-port METRICS_AGENT_PORT
(raylet)                          --runtime-env-agent-port RUNTIME_ENV_AGENT_PORT
(raylet)                          [--object-spilling-config OBJECT_SPILLING_CONFIG]
(raylet)                          [--logging-rotate-bytes LOGGING_ROTATE_BYTES]
(raylet)                          [--logging-rotate-backup-count LOGGING_ROTATE_BACKUP_COUNT]
(raylet)                          [--runtime-env-hash RUNTIME_ENV_HASH] --startup-token
(raylet)                          STARTUP_TOKEN [--ray-debugger-external]
(raylet)                          [--session-name SESSION_NAME] [--webui WEBUI]
(raylet)                          --worker-launch-time-ms WORKER_LAUNCH_TIME_MS
(raylet)                          [--worker-preload-modules WORKER_PRELOAD_MODULES]
(raylet) default_worker.py: error: the following arguments are required: --startup-token, --worker-launch-time-ms
```

I investigated using strace and got arguments which are passed to default_worker.py:
```
[pid 17910] +++ exited with 0 +++
[pid 17815] execve("/usr/bin/bash", ["bash", "-c", "exec /home/ray/anaconda3/bin/python /home/ray/anaconda3/lib/python3.8/site-packages/ray/_private/workers/default_worker.py --node-ip-address=192.168.0.2 --node-manager-port=42547 --object-store-name=/tmp/ray/session_2023-12-01_13-42-35_650605_16416/sockets/plasma_store --raylet-name=/tmp/ray/session_2023-12-01_13-42-35_650605_16416/sockets/raylet --redis-address=None --temp-dir=/tmp/ray --metrics-agent-port=59796 --runtime-env-agent-port=62111 --logging-rotate-bytes=536870912 --logging-rotate-backup-count=5 --runtime-env-agent-port=62111 --gcs-address=192.168.0.2:6379 --session-name=session_2023-12-01_13-42-35_650605_16416 --temp-dir=/tmp/ray --webui=192.168.0.2:8265 --cluster-id=e0269812994822291b61b2be7bc4a045f747c265f98712c687ce2dc3 --storage=s3://explore/ray?&endpoint_override=http://192.168.0.4:9000 --startup-token=16 --worker-launch-time-ms=1701466993176 --runtime-env-hash=1559810005"], 0x101b950 /* 71 vars */) = 0
strace: Process 17922 attached
[pid 17922] execve("/home/ray/anaconda3/bin/python", ["/home/ray/anaconda3/bin/python", "/home/ray/anaconda3/lib/python3.8/site-packages/ray/_private/workers/default_worker.py", "--node-ip-address=192.168.0.2", "--node-manager-port=42547", "--object-store-name=/tmp/ray/session_2023-12-01_13-42-35_650605_16416/sockets/plasma_store", "--raylet-name=/tmp/ray/session_2023-12-01_13-42-35_650605_16416/sockets/raylet", "--redis-address=None", "--temp-dir=/tmp/ray", "--metrics-agent-port=59796", "--runtime-env-agent-port=62111", "--logging-rotate-bytes=536870912", "--logging-rotate-backup-count=5", "--runtime-env-agent-port=62111", "--gcs-address=192.168.0.2:6379", "--session-name=session_2023-12-01_13-42-35_650605_16416", "--temp-dir=/tmp/ray", "--webui=192.168.0.2:8265", "--cluster-id=e0269812994822291b61b2be7bc4a045f747c265f98712c687ce2dc3", "--storage=s3://explore/ray?"], 0x561ebe4434f0 /* 71 vars */strace: Process 17923 attached
) = 0
[pid 17923] +++ exited with 127 +++

```

It seems that equal sign in  --storage="s3://explore/ray?&endpoint_override**=**http://192.168.0.4:9000"` parameter is not escaped before passing to default_worker.py and it causes  that

### Versions / Dependencies

Tested ray version 2.7.1  and 2.8.1

### Reproduction script

I provided all scripts above

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by tvildo**
fixes ray-project/ray#41568

## Why are these changes needed?

when starting ray using --storage argument like that:
```
ray start --head  --dashboard-host=0.0.0.0 --block --storage="s3://explore/ray?&endpoint_override=http://192.168.0.3:9000"
```
Raylet on worker will crash

## Related issue number

Closes #41568

### CODE REVIEW

**Comment by tvildo** [comment]
@jjyao  All tests are passing now.

File: python/ray/_private/services.py [hunk 1]
```diff
@@ -1676,7 +1676,7 @@ def start_raylet(
     )
 
     if storage is not None:
-        start_worker_command.append(f"--storage={storage}")
+        start_worker_command.append(f'--storage="{storage}"')
```
Comment (jjyao) [inline python/ray/_private/services.py:1679]: I think if you do `f"--storage='{stroage}'"`, you may not need the `strip` change.

**Comment by tvildo** [comment]
@jjyao  I tried with single quote and it still needs **strip** https://buildkite.com/ray-project/premerge/builds/15005 I am reverting back my changes

**Comment by jjyao** [comment]
I see. So it's a windows only issue? I saw tests passed for linux.

**Comment by tvildo** [comment]
Yes. With **strip** it works in all environments.

**Comment by tvildo** [comment]
@jjyao Can I do anything else to be able to merge this pull request ?

**Comment by girogisturua** [comment]
I have same issue, great solution, please, merge asap

**Comment by jjyao** [comment]
@tvildo I did more investigation and found the root cause and have a fix here: https://github.com/ray-project/ray/pull/42332. Could you try it out and see if it fixes your issue?

**Comment by anyscalesam** [comment]
closing as old here per @jjyao above should fix this already - @tvildo can you confirm/feel to reopen if disagree.

**Pull request closed without merging**